tools/pyboard.py

```diff
@@ -545,7 +546,11 @@ def main():
 
         def execbuffer(buf):
             try:
-                ret, ret_err = pyb.exec_raw(buf, timeout=None, data_consumer=stdout_write_bytes)
+                if args.no_follow:
+                    pyb.exec_raw_no_follow(buf)
+                    ret_err = None
+                else:
+                    ret, ret_err = pyb.exec_raw(buf, timeout=None, data_consumer=stdout_write_bytes)
```

Comment:
Using `no_follow` here implements more of a "detach early" operation: in normal situations the script is piped to the board, run, and then pyboard.py waits for it to finish running.  The `no_follow` option here is just forcing it to do the first part of that sequence, ie download script and run, then detach.

So it might be better to call the option `--detach-early`.... but that's possibly confusing and in the end `--no-follow` is more logical because it corresponds to `exec_raw` vs `exec_raw_no_follow`.

In summary I'm ok with this part of the PR.